app/uk/gov/hmrc/agentuserclientdetails/services/AssignedUsersService.scala

```diff
@@ -40,20 +41,58 @@ trait AssignedUsersService {
 class AssignedUsersServiceImpl @Inject() (
   es3CacheManager: Es3CacheManager,
   espConnector: EnrolmentStoreProxyConnector,
+  serviceInstances: ServiceInstances,
   appConfig: AppConfig
-)(implicit ec: ExecutionContext, materializer: Materializer)
+)(implicit ec: ExecutionContext, actorSystem: ActorSystem)
     extends AssignedUsersService with Logging {
 
-  logger.info(s"ES0 requests set to throttle at ${appConfig.es0MaxRequestsPerSecond} per second")
+  logger.info(s"ES0 requests set to throttle at ${appConfig.es0ThrottlingRate}")
+
+  lazy val es0Throttler: ThrottledWorkItemProcessor =
+    new ThrottledWorkItemProcessor(
+      "es0-fetch-assigned-users",
+      actorSystem,
+      rateLimit = Some(Rate.parse(appConfig.es0ThrottlingRate))
+    ) {
+      def instanceCount: Int = Option(serviceInstances).fold(1)(
+        _.instanceCount
+      ) // must handle serviceInstances == null case (can happen in testing)
+    }
 
   override def calculateClientsWithAssignedUsers(
     groupId: String
   )(implicit hc: HeaderCarrier): Future[Seq[AssignedClient]] =
     for {
-      clients         <- es3CacheManager.getCachedClients(groupId)
-      assignedClients <- Throttler.process(clients, appConfig.es0MaxRequestsPerSecond)(fetchAssignedUsersOfClient)
+      allClients      <- es3CacheManager.getCachedClients(groupId)
+      assignedClients <- accumulateAssignedClients(allClients)
     } yield assignedClients
 
+  private def accumulateAssignedClients(
+    clients: Seq[Client]
+  )(implicit hc: HeaderCarrier): Future[Seq[AssignedClient]] = {
+    val clientBatches = clients.grouped(20)
+
+    clientBatches.foldLeft(Future successful Seq.empty[AssignedClient]) { (previousFuture, clientBatch) =>
+      for {
+        accumulated     <- previousFuture
+        assignedClients <- fetchAssignedUsersOfClientBatch(clientBatch)
+      } yield accumulated ++ assignedClients
+    }
+  }
+
+  private def fetchAssignedUsersOfClientBatch(
+    clients: Seq[Client]
+  )(implicit hc: HeaderCarrier): Future[Seq[AssignedClient]] =
+    Future
+      .traverse(clients) { client =>
+        if (appConfig.enableThrottling) {
+          es0Throttler.throttledStartingFrom(DateTime.now())(
```

Comment:
I am not sure this will do what you expect, I think it will send out requests 20 times faster than you want - as all those from a group of 20 will probably be sent around the same time